docs: port push notification guide (#537)
GetStream/stream-chat-flutter: docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx

```dart
firebaseMessaging.onTokenRefresh.listen((token) { 
      client.addDevice(token, PushProvider.firebase); 
}); 
```

### Possible issues

We only send push notifications when the user doesn't have any active websocket connection (which is established when you call `client.connectUser`). If you set the [onBackgroundEventReceived](https://pub.dev/documentation/stream_chat_flutter/latest/stream_chat_flutter/StreamChat/onBackgroundEventReceived.html) property of the StreamChat widget, when your app goes to background, your device will keep the ws connection alive for 1 minute, and so within this period, you won't receive any push notification.

Make sure to read the [general push docs](https://getstream.io/chat/docs/flutter-dart/push_introduction/?language=dart) in order to avoid known gotchas that may make your relationship with notifications go bad 😢

### Testing if Push Notifications are Setup Correctly

> If you're not sure if you've set up push notifications correctly (eg you don't always receive them, they work unreliably), you can follow these steps to make sure your config is correct and working:

Reviewer: Small fix

```suggestion
If you're not sure if you've set up push notifications correctly (e.g. you don't always receive them, they work unreliably), you can follow these steps to make sure your config is correct and working:
```

:::

#### Step 6

Save your push notification settings changes

![](../assets/dashboard_save_changes.jpeg)

**OR**

Upload the `Server Key` via API call using a backend SDK

```js
await client.updateAppSettings({ 
    firebase_config: { 
        server_key: 'server_key', 
        notification_template: `{"message":{"notification":{"title":"New messages","body":"You have {{ unread_count }} new message(s) from {{ sender.name }}"},"android":{"ttl":"86400s","notification":{"click_action":"OPEN_ACTIVITY_1"}}}}`, 
        data_template: `{"sender":"{{ sender.id }}","channel":{"type": "{{ channel.type }}","id":"{{ channel.id }}"},"message":"{{ message.id }}"}` 
    }, 
}); 
```

### Registering a device at Stream Backend

Once you configure Firebase server key and set it up on Stream dashboard a device that is supposed to receive push notifications needs to be registered at Stream backend. This is usually done by listening for Firebase device token updates and passing them to the backend as follows:

```dart
firebaseMessaging.onTokenRefresh.listen((token) { 
      client.addDevice(token, PushProvider.firebase); 
}); 
```

### Possible issues

> We only send push notifications, when the user doesn't have any active websocket connection (which is established when you call client.connectUser). If you set the [onBackgroundEventReceived](https://pub.dev/documentation/stream_chat_flutter/latest/stream_chat_flutter/StreamChat/onBackgroundEventReceived.html) property of the StreamChat widget, when your app goes to background, your device will keep the ws connection alive for 1 minute, and so within this period, you won't receive any push notification.

Reviewer: Unnecessary comma

> We only send push notifications, when the user doesn't have any active websocket connection (which is established when you call client.connectUser). If you set the [onBackgroundEventReceived](https://pub.dev/documentation/stream_chat_flutter/latest/stream_chat_flutter/StreamChat/onBackgroundEventReceived.html) property of the StreamChat widget, when your app goes to background, your device will keep the ws connection alive for 1 minute, and so within this period, you won't receive any push notification.

Reviewer: Unnecessary comma and maybe add a possible highlight

```suggestion
We only send push notifications when the user doesn't have any active websocket connection (which is established when you call `client.connectUser`). If you set the [onBackgroundEventReceived](https://pub.dev/documentation/stream_chat_flutter/latest/stream_chat_flutter/StreamChat/onBackgroundEventReceived.html) property of the StreamChat widget, when your app goes to background, your device will keep the ws connection alive for 1 minute, and so within this period, you won't receive any push notification.
```

          EventType.notificationMessageNew, 
        ].contains(event.type) || 
        event.user.id == currentUserId) { 
      return; 
    } 
    if (event.message == null) return; 
    final flutterLocalNotificationsPlugin = FlutterLocalNotificationsPlugin(); 
    final initializationSettingsAndroid = 
        AndroidInitializationSettings('launch_background'); 
    final initializationSettingsIOS = IOSInitializationSettings(); 
    final initializationSettings = InitializationSettings( 
      android: initializationSettingsAndroid, 
      iOS: initializationSettingsIOS, 
    ); 
    await flutterLocalNotificationsPlugin.initialize(initializationSettings); 
    await flutterLocalNotificationsPlugin.show( 
      event.message.id.hashCode, 
      event.message.user.name, 
      event.message.text, 
      NotificationDetails( 
        android: AndroidNotificationDetails( 
          'message channel', 
          'Message channel', 
          'Channel used for showing messages', 
          priority: Priority.high, 
          importance: Importance.high, 
        ), 
        iOS: IOSNotificationDetails(), 
      ), 
    ); 
  }, 
  child: .... 
); 
... 
```

> As you can see we §generate a local notification whenever a message.new or notification.message_new event is received.

Reviewer: Is the '§' doing something or symbolic of something? Remove if a typo.

Author: lol that's a typo 
```suggestion
As you can see we generate a local notification whenever a message.new or notification.message_new event is received.
```
